tests/test_collections.py

```diff
@@ -345,6 +345,38 @@ def test_feature_collection_with_dates_serializes_correctly():
         assert fc[0].attributes == expected_attributes
 
 
+class GeoFeatureThatProduced257():
```

Comment:
Why not

```python
bounds = GeoVector.from_bounds(xmin=-295505.5513504914,
                               ymin=4705921.785461731,
                               xmax=-295352.3793504914,
                               ymax=4706074.957461731,
                               crs={'init': 'epsg:3857'})
resolution = 0.596
raster = GeoRaster2.empty_from_roi(roi=bounds, resolution=resolution)
with tempfile.NamedTemporaryFile(suffix='.json') as fp:
    raster.save(fp.name)
    feature = GeoFeature(raster.footprint(), {"raster_url": fp.name})
    fc = FeatureCollection([feature])
    tile = fc.get_tile(z=18, x=129139, y=100288)
    assert tile.shape == (1, 256, 256)
```

?

Reply:
I think the code I worte is more reusable in case you want that raster in a different test

Reply:
I still think the current code is too complex, but if the objective is raster reusability, I'd reuse only the raster. Something like:

```python
class NamedTemporaryRaster(tempfile.NamedTemporaryFile):
    def _init_raster(self):
        bounds = GeoVector.from_bounds(xmin=-295505.5513504914,
                                       ymin=4705921.785461731,
                                       xmax=-295352.3793504914,
                                       ymax=4706074.957461731,
                                       crs={'init': 'epsg:3857'})
        resolution = 0.596
        return GeoRaster2.empty_from_roi(roi=bounds, resolution=resolution)

    def __enter__(self):
        raster = self._init_raster()
        raster.save(self.name)
        return raster

with NamedTemporaryRaster() as rs:
    feature = GeoFeature(rs.footprint(), {"raster_url": rs._filename})
    fc = FeatureCollection([feature])
    tile = fc.get_tile(z=18, x=129139, y=100288)
    assert tile.shape == (1, 256, 256)
```

Reply:
I'm missing a call to `super().__enter__()` somewhere, but I think I made the point